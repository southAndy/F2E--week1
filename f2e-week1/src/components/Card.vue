<template>
  <router-link :to="{ name: 'Content' }" class="card">
    <div class="card_image">
      <img
        v-if="parentData.Picture.PictureUrl1"
        :src="parentData.Picture.PictureUrl1"
        alt="活動照片"
      />
      <img v-else src="@/assets/image/search.svg" alt="該資料沒有圖片顯示" />
    </div>
    <div class="card_body">
      <div class="time">
        {{ convertStartTime }} -
        {{ convertEndTime }}
      </div>
      <p class="title">{{ parentData.ActivityName }}</p>
      <a href="##" class="location">{{ parentData.City }}</a>
    </div>
  </router-link>
</template>

<script>
export default {
  name: "Card",
  props: ["activitiesData"],
  data() {
    return {
      parentData: this.activitiesData,
      //todo 亂數取得
      // randomNumber: Number(),
    };
  },
  methods: {
    getRandom() {},
  },
  computed: {
    convertStartTime() {
      let activitiesTime = new Date(this.parentData.StartTime);
      let activityYear = activitiesTime.getFullYear();
      let activityMonth = activitiesTime.getMonth();
      let activityDate = activitiesTime.getDate();
      console.log(activityYear, activityMonth, activityDate);
      return `${activityYear}/${activityMonth}/${activityDate}`;
    },
    convertEndTime() {
      let activitiesTime = new Date(this.parentData.EndTime);
      let activityYear = activitiesTime.getFullYear();
      let activityMonth = activitiesTime.getMonth();
      let activityDate = activitiesTime.getDate();
      console.log(activityYear, activityMonth, activityDate);
      return `${activityYear}/${activityMonth}/${activityDate}`;
    },
  },
};
</script>
<style lang="scss" scoped>
@use "../assets/sass/breakpoints.scss";

.card {
  display: flex;

  width: 345px;
  height: 62px;

  text-decoration: none;
  //card's distance
  margin-bottom: 16px;

  &_image {
    width: 100px;
    height: 100%;

    img {
      width: 100%;
      object-fit: contain;

      border-radius: 10px;
      border: 1px solid;
    }
  }
  &_body {
    width: 354px- 90px;
    height: 100%;

    text-align: start;

    overflow: hidden;
    margin-left: 20px;

    .title {
      font-weight: 700;
      font-size: 18px;
      color: black;

      margin: 0;
      @include breakpoints.tablet {
        font-size: 20px;
      }
    }
    .time {
      font-size: 14px;
      text-decoration: none;
      color: #646464;
    }
    .location {
      font-size: 14px;
      font-weight: 400;
      text-decoration: none;
      color: #646464;
    }
  }
}
</style>
