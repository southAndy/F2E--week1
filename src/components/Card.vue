<template>
  <router-link
    :to="{
      name: 'Content',
      params: {
        city: activities?.City,
        id: activities?.ActivityID,
        name: activities?.ActivityName,
      },
    }"
    class="card"
    :class="{ skeleton: this.$store.state.isLoading }"
  >
    <div class="card_image">
      <img
        v-if="activities?.Picture?.PictureUrl1"
        :src="activities?.Picture?.PictureUrl1"
        alt="活動照片"
      />
    </div>
    <div class="card_body">
      <div class="time">
        {{ convertStartTime??'' }} -
        {{ convertEndTime??'' }}
      </div>
      <p class="title">{{ activities?.ActivityName }}</p>
      <i></i>
      <a v-if="activities.City" href="##" class="location">
        <img src="@/assets/image/Vector.png" alt="座標圖示" />
        {{ activities?.City }}
      </a>
    </div>
    <p class="card_more">詳細介紹</p>
  </router-link>
</template>

<script>
export default {
  name: "Card",
  props: ["activitiesData"],
  methods: {
    // getRandom() {},
  },
  computed: {
    convertStartTime() {
      if(this.activities.StartTime){
        let activitiesTime = new Date(this.activities.StartTime);
        let activityYear = activitiesTime.getFullYear();
        let activityMonth = activitiesTime.getMonth();
        let activityDate = activitiesTime.getDate();
        // console.log(activityYear, activityMonth, activityDate);
        return `${activityYear}/${activityMonth}/${activityDate}`;

      }
    },
    convertEndTime() {
      if(this.activities.EndTime){
        let activitiesTime = new Date(this.activities.EndTime);
        let activityYear = activitiesTime.getFullYear();
        let activityMonth = activitiesTime.getMonth();
        let activityDate = activitiesTime.getDate();
        // console.log(activityYear, activityMonth, activityDate);
        return `${activityYear}/${activityMonth}/${activityDate}`;

      }
    },
  },
};
</script>
<style lang="scss" scoped>
@use "../assets/sass/breakpoints.scss";
@use "../assets/sass/loading.scss";
.skeleton {
  @extend %skeleton-loading;
}

.card {
  display: flex;
  flex-direction: row;
  width: 345px;
  height: 70px;

  text-decoration: none;
  //card's distance
  margin-bottom: 16px;

  border: none;
  @include breakpoints.tablet {
    flex-direction: row;
    width: 45%;
    background-color: #f9f9f9;
    margin-right: 30px;
    // border: 1px solid #646464;
    border-radius: 16px;

    &:hover {
      background: linear-gradient(to right, #f9f9f9, #7f977b);
    }
  }
  @include breakpoints.desktop {
    height: 160px;
  }

  &_image {
    width: 100px;
    height: 100%;
    @include breakpoints.desktop {
      flex: 1;
      width: 160px;
      height: 160px;
    }

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;

      border-radius: 10px;
      border: 1px solid;
    }
  }
  &_body {
    width: 354px- 90px;
    height: 100%;

    text-align: start;

    overflow: hidden;
    margin-left: 20px;
    @include breakpoints.desktop {
      flex: 1.5;
      padding: 16px 0;
    }

    .title {
      font-weight: 700;
      font-size: 17px;
      color: black;

      margin: 0;
      @include breakpoints.tablet {
        font-size: 15px;
        height: 25px;
        overflow: hidden;
      }
      @include breakpoints.desktop {
        font-size: 19px;
      }
    }
    .time {
      font-size: 14px;
      text-decoration: none;
      color: #646464;
      @include breakpoints.tablet {
        font-size: 16px;
      }
    }
    .location {
      display: inline-block;
      font-size: 14px;
      font-weight: 400;
      text-decoration: none;
      color: #646464;
      @include breakpoints.tablet {
        font-size: 16px;
        margin-top: 55px;
      }
    }
  }
  &_more {
    display: none;
    @include breakpoints.desktop {
      display: block;
      color: rgb(153, 150, 150);
      padding: 0 10px;
      align-self: end;
    }
  }
}
</style>
