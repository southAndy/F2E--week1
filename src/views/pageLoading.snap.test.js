import { shallowMount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import Card from '@/components/Card.vue';

describe('Card.vue', () => {
  
  // 模擬類似真實 api 回傳的假資料
  const activitiesData = {
    StartTime: '2024-11-28T12:00:00Z',
    EndTime: '2024-11-29T12:00:00Z',
    City: 'Taipei',
    ActivityID: '12345',
    ActivityName: 'Sample Activity',
    Picture: { PictureUrl1: 'https://example.com/sample.jpg' },
  };

  // 模擬的 vuex
  const mockStore = {
    state: {
      isLoading: false,
    },
  };

  it('renders correctly when not loading', () => {
    const wrapper = shallowMount(Card, {
      props: { activitiesData },
      global: {
        mocks: {
          $store: mockStore, // 模擬 Vuex
        },
        stubs: {
          'router-link': true, // 模擬 router-link
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders skeleton class when loading', () => {
    mockStore.state.isLoading = true; // 模擬 isLoading 狀態
    const wrapper = shallowMount(Card, {
      props: { activitiesData },
      global: {
        mocks: {
          $store: mockStore,
        },
        stubs: {
          'router-link': true,
        },
      },
    });

    expect(wrapper.find('.skeleton').exists()).toBe(true); // 檢查 skeleton class 是否存在
  });
});