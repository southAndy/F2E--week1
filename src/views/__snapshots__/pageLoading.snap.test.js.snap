// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card.vue > renders correctly when not loading 1`] = `"<router-link-stub data-v-c6c3362a="" to="[object Object]" class="card"></router-link-stub>"`;

exports[`Card.vue > renders correctly with Vuex and activitiesData 1`] = `"<router-link-stub data-v-c6c3362a="" to="[object Object]" class="card"></router-link-stub>"`;
